src/CodeFormatter.cs

```diff
@@ -247,6 +258,16 @@ await GeneratedCodeUtilities.IsGeneratedCodeAsync(syntaxTree, cancellationToken)
                         documentsNotCoveredByEditorConfig.Add(document.Id);
                     }
                 }
+
+                if (formatOptions.IncludeGeneratedFiles)
+                {
+                    var generatedDocuments = await project.GetSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false);
+                    foreach (var generatedDocument in generatedDocuments)
+                    {
+                        Debug.WriteLine($"Including source generated file '{generatedDocument.FilePath}'.");
+                        sourceGeneratedDocuments.Add(generatedDocument.Id);
```

Comment:
Rather than passing IDs around, is it easier to just pass the Document objects around directly?